src/renderer/views/AppSettings/AppSettings.tsx

```diff
@@ -1,6 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './AppSettings.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { Button, MenuItem, Select } from '@mui/material';
+import {
+  selectCurrentLocale,
+  setLocale,
+} from '../../../shared/redux/slices/settingsSlice';
 
-const AppSettings = () => <div>AppSettings</div>;
+const AppSettings = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const [language, setLanguage] = useState(useSelector(selectCurrentLocale));
+
+  function submitChanges() {
+    dispatch(setLocale(language));
+  }
+
+  return (
+    <div>
+      {t('AppSettings.title')}
+      <Select
+        label={t('AppSettings.language')}
+        value={language}
+        onChange={(e) => setLanguage(e.target.value as 'en' | 'pl')}
+      >
+        <MenuItem value='en'>{t('AppSettings.language.english')}</MenuItem>
+        <MenuItem value='pl'>{t('AppSettings.language.polish')}</MenuItem>
```

Comment:
Similar future proofing could be done here by importing the `resources` from `i18next.ts` (because i18n does not provide a list of all supported languages unfortunately; it will bait you with `i18n.languages` but that's not all the available locales) and mapping over the keys of that object to generate menu items. With that, the key of translation could change to `AppSettings.language.pl` so that you can use the `resources` object key to retrieve the translation.

It could perhaps be even better to have a `supportedLocales` object exported from the `i18next.ts`. Something like: 
```ts
export const supportedLocales = Object.keys(resources) as SupportedLangCode[];
```

Then you could map over that instead of importing the whole `resources`.

Reply:
done